DeepSpeech.py

```diff
@@ -307,8 +307,26 @@ def get_tower_results(iterator, optimizer, dropout_rates):
                     tower_avg_losses.append(avg_loss)
 
                     # Compute gradients for model parameters using tower's mini-batch
-                    gradients = optimizer.compute_gradients(avg_loss)
-
+                    if FLAGS.load == "transfer":
+                        if not FLAGS.fine_tune:
+                            # train from source model and freeze old layers
+                            # aka - only update new layers
+                            gradients = optimizer.compute_gradients(
+                                avg_loss,
+                                var_list = [ v for v in tfv1.trainable_variables()
+                                             if any(
+                                                     layer in v.op.name
+                                                     for layer in drop_source_layers
+                                             )]
+                            )
+                        else:
+                            # Transfer learning, but update all layers
+                            # aka - fine tune all layers
+                            gradients = optimizer.compute_gradients(avg_loss)
+                    else:
+                        # no Transfer Learning, and updating all layers 
+                        gradients = optimizer.compute_gradients(avg_loss)
```

Comment:
I don't get why you need the additional "transfer" option of the "load" flag. Why not just relying on the "fine_tune" flag?
In this case you now got two times `gradients = optimizer.compute_gradients(avg_loss)`. One time for the `FLAGS.load == "transfer" and FLAGS.fine_tune` case and the other time for the `FLAGS.load != "transfer"` case.

Reply:
Actually, you have a good point here, @tilmankamp 

Up to now, we have "starting from an old checkpoint" separate from "transfer learning". However, transfer-learning is a special case of starting from an old checkpoint.

maybe we should combine these two features, so that the user can remove any number of source layers for transfer, *including* removing zero layers.

thoughts?